subsys/usb/usb_device.c

```diff
@@ -804,7 +804,10 @@ static bool usb_handle_std_interface_req(struct usb_setup_packet *setup,
 		return false;
 
 	case REQ_GET_INTERFACE:
-		/* there is only one interface, return n-1 (= 0) */
+		/* Interfaces that have alternates are handled by classes code.
```

Comment:
Tbh I am happy You wrote that. I think I have an idea of how We can accomplish that. Instead of `usb_cfg_data::const usb_if_descriptor *const *const interfaces`

Reply:
Tbh I am happy You mentioned that. I think I have an idea of how We can accomplish that.
Instead of `usb_cfg_data :: interfaces` with type of `const struct usb_if_descriptor *const *const` We can create a list of structs which not only points to interface but to alternate (if exists) as well + whatever we want to have there. Having that we will be able to "know" the configured function/class form core.

Simply speaking something like:
`const struct interface_container *const *const` 
Where `struct interface_container` is:
```
struct interface_container {
    struct usb_if_descriptor *iface;
    struct usb_if_descriptor *iface_alt;
    uint8_t curr_alt;
};
```